File: HW1/Problem5/insertsort.py
from random import randint
import timeit

def genRandomNumbers(numToGen):
	numArray = []

	for x in xrange(0,numToGen):
		numArray.append(randint(0,10000))

	return numArray

def insertionSort(array):
	for j in xrange(1,len(array)):
		key = array[j]

		i = j - 1
		while (i > -1) and (array[i] > key):
			array[i + 1] = array[i]
			i = i - 1

		array[i + 1] = key

timeArray = (1000,2000,5000,10000,20000,30000,35000)

outFile = open("inserttime.csv", "w")

outFile.write("num,time\n")

for num in timeArray:
	for x in xrange(0,5):
		randomArray = genRandomNumbers(num)
		start_time = timeit.default_timer()
		insertionSort(randomArray)
		elapsed = timeit.default_timer() - start_time

		outFile.write(str(num) + "," + str(elapsed) + '\n')

outFile.close()

File: HW1/Problem5/insertsort.js
"use strict";

const fs = require("fs");

const getRandomInt = (min, max) => {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min)) + min;
};

const genRandomNumbers = numToGen => {
    let numArray = [];

    for(let i = 0; i < numToGen; i++) {
        numArray.push(getRandomInt(0,10000));
    }

    return numArray;
};

const insertionSort = array => {
    for(let j = 1; j < array.length; j++) {
        const key = array[j];

        let i = j - 1;
        while((i > -1) && (array[i] > key)) {
            array[i + 1] = array[i];
            i = i - 1;
        }

        array[i + 1] = key;
    }
};

const timeArray = [500,1000,1500,2000,2500,3000,3500];;

fs.writeFileSync("inserttime.csv", "num,time\n");

timeArray.forEach(num => {
    for(let i = 0; i < 5; i++) {
        const randomArray = genRandomNumbers(num);
        const start = new Date();
        const hrstart = process.hrtime();
        insertionSort(randomArray);
        const end = new Date() - start;
        const hrend = process.hrtime(hrstart);
        fs.appendFileSync("inserttime.csv", num + "," + end + "\n");
    }
});
